library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -232,48 +178,46 @@ impl<T, A: Allocator> VecDeque<T, A> {
     /// Returns `true` if the buffer is at full capacity.
     #[inline]
     fn is_full(&self) -> bool {
-        self.cap() - self.len() == 1
+        self.len == self.capacity()
     }
 
     /// Returns the index in the underlying buffer for a given logical element
-    /// index.
+    /// index + addend.
     #[inline]
-    fn wrap_index(&self, idx: usize) -> usize {
-        wrap_index(idx, self.cap())
+    fn wrap_add(&self, idx: usize, addend: usize) -> usize {
+        wrap_index(idx.wrapping_add(addend), self.capacity())
     }
 
-    /// Returns the index in the underlying buffer for a given logical element
-    /// index + addend.
     #[inline]
-    fn wrap_add(&self, idx: usize, addend: usize) -> usize {
-        wrap_index(idx.wrapping_add(addend), self.cap())
+    fn wrap_idx(&self, idx: usize) -> usize {
```

Comment:
nit: I see from line 188 that there's also a `wrap_index` function in use here, so please don't have `wrap_index` and `wrap_idx` -- if `wrap_idx` is the same conceptual operation as `wrap_index`, just as a method, call it `wrap_index`, and if it's something different please give it a more clearly-different name.

Reply:
Hmm, as I read more, maybe this is `logical_index_to_physical`?

Reply:
Yes, the name is maybe a bit unfortunate but it just converts logical to physical indices. I guess I'll rename it to something like `to_physical_idx`.